commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -138,48 +139,122 @@ private void rangeCheckForInsert(int index) {
 
     /**
      * Gets the complex number \( (a + i b) \) at the indexed position of the list.
-     *
+     * <p>
      * {@inheritDoc}
+     *
      * @return the complex number.
      */
     @Override
     public Complex get(int index) {
         rangeCheck(index);
-        final int i = index << 1;
-        return Complex.ofCartesian(realAndImagParts[i], realAndImagParts[i + 1]);
+        return Complex.ofCartesian(realAndImagParts[index << 1], realAndImagParts[(index << 1) + 1]);
+    }
+
+    /**
+     * Gets the real part \( a \) of the complex number \( (a + i b) \) at the indexed position of the list.
+     *
+     * @param index Index of the complex number.
+     * @return The real part.
+     */
+    public double getReal(int index) {
+        rangeCheck(index);
+        return realAndImagParts[index << 1];
+    }
+
+    /**
+     * Gets the imaginary part \( b \) of the complex number \( (a + i b) \) at the indexed position of the list.
+     *
+     * @param index Index of the complex number.
+     * @return The imaginary part.
+     */
+    public double getImaginary(int index) {
+        rangeCheck(index);
+        return realAndImagParts[(index << 1) + 1];
     }
 
     /**
-     * Replaces the element at the specified position in this list with the specified element's
+     * Replaces the complex number at the specified position in the list with the specified complex number's
      * real and imaginary parts. No range checks are done.
      *
-     * @param index Index of the element to replace.
+     * @param index Index of the complex number to replace.
      * @param real Real part \( a \) of the complex number \( (a +ib) \).
      * @param imaginary Imaginary part \( b \) of the complex number \( (a +ib) \).
      */
     private void setNoRangeCheck(int index, double real, double imaginary) {
-        final int i = index << 1;
-        realAndImagParts[i] = real;
-        realAndImagParts[i + 1] = imaginary;
+        realAndImagParts[index << 1] = real;
+        realAndImagParts[(index << 1) + 1] = imaginary;
     }
 
     /**
      * {@inheritDoc}
      *
-     * @param element Complex element to be set.
+     * @param number Complex number to be set.
      */
     @Override
-    public Complex set(int index, Complex element) {
+    public Complex set(int index, Complex number) {
         rangeCheck(index);
-        final int i = index << 1;
```

Comment:
The precomputed value i will be reused so it can be left as before, please revert.

In addition the call to setNoRangeCheck can be removed. Since you already have the value i this will avoid calling it again.